stackrox/stackrox
# ROX-22031: Return status error on missing Compliance permissions

## Description

Perform a SelfSubjectAccessReview for Sensor to indicate whether it has write or read access. When `allowed: false` is returned we can assume read access because of Sensors `stackrox:view-cluster` role.
Sensor in this case writes the error to the `statusErrors` field.

API integrations response:

```
# Compliance role is installed, curl endpoint
$ roxcurl /v2/compliance/integrations
{"integrations":[{"id":"b09ae500-c67e-47d5-9994-2049be321ca7","version":"","clusterId":"53f4e2f2-f64c-4d2f-b85a-7f3993ff5e49","clusterName":"remote","namespace":"openshift-compliance","statusErrors":[]}]}


# Compliance role is removed, curl endpoint
$ roxcurl /v2/compliance/integrations
{"integrations":[{"id":"b09ae500-c67e-47d5-9994-2049be321ca7","version":"","clusterId":"53f4e2f2-f64c-4d2f-b85a-7f3993ff5e49","clusterName":"remote","namespace":"","statusErrors":["Sensor cannot write compliance.openshift.io API group resources. Please check Sensor's RBAC permissions."]}]}
```

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

 - Install StackRox + Compliance Operator via `./deploy/openshift/deploy.sh`
 - Check Sensor has write access
   - Run `roxcurl /v2/compliance/integrations` and check that `readOnly: false` is returned
 - Check Sensor reports read access
   - Run `kubectl edit clusterrole stackrox:edit-compliance` and modify the verbs from `*` to `get`
   - Run `watch -n3 roxcurl /v2/compliance/integrations` until `readOnly: false`.
